### PR TITLE
Add GitHub Action for Tor update check

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review

Refs https://github.com/freedomofpress/securedrop-builder/issues/482

The `new-tor-issue` script and parts of the job code were imported from the `securedrop-builder` repo and authored by Kunal Mehta, see https://github.com/freedomofpress/securedrop-builder/pull/408 and https://github.com/freedomofpress/securedrop-builder/pull/287

## Description of changes

Migrate logic for checking for and applying Tor updates to GitHub Actions

### CODE REVIEW

File: .github/workflows/update-tor.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Tor package updates
+on:
+  schedule:
+    - cron: "0 6 * * *"
+jobs:
+    check-for-tor-updates:
+        runs-on: ubuntu-latest
+        container: debian:bullseye-backports
```
Comment (eloquence) [inline .github/workflows/update-tor.yml:8]: I copied this from the CircleCI job, but I did not research whether the `backports` image specifically is still needed, or if we should switch to a different one.

**Comment by eloquence** [comment]
I think this is ready for a review pass. What I've tested:
- automatic issue creation (to a sandbox repo, via a test token)
- automatic pushes (to a test branch, via the existing PUSH_TOKEN)

Using both tokens in the same job seems to work fine, since `ISSUE_TOKEN` is only needed for the `gh` commands, which use the distinct `GH_TOKEN` variable.

I've not tested the actual `reprepo` behavior aside for looking out for errors in the job output. Here's an example run:

https://github.com/freedomofpress/securedrop-apt-test/actions/runs/8430760453/job/23087122497?pr=221

I'm happy to help with any additional pre-merge testing. https://github.com/freedomofpress/securedrop-builder/pull/493 removes the corresponding CircleCI logic in `securedrop-builder` once this one is ready (marked draft for now to avoid accidental merge).

**Review by legoktm — CHANGES_REQUESTED** [review]
Overall this looks good, I think the testing you've done is sufficient to land this, if there's an issue with permissions or whatever we can fix it iteratively.

File: scripts/new-tor-issue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+to any existing issues
+"""
+
+import json
+import random
+import subprocess
+import tempfile
+
+# FIXME: Replace with freedomofpress/securedrop once everything works
+REPOSITORY = "freedomofpress/securedrop-issues-sandbox"
```
[the remark this answers is not in the document]
Comment (eloquence) [reply scripts/new-tor-issue]: Whoops! That's a bit embarrassing. Thanks.

File: .github/workflows/update-tor.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: Tor package updates
+on: [pull_request]
```
Comment (legoktm) [inline .github/workflows/update-tor.yml:2]: Sorry I missed this earlier. This job should be timer based so it runs daily and preferably not at the same time SDW nightlies run (6am per https://github.com/freedomofpress/securedrop-client/blob/main/.github/workflows/nightlies.yml#L4)...so how about 7am?

File: .github/workflows/update-tor.yml (hunk 1 above)
Comment (eloquence) [reply .github/workflows/update-tor.yml]: Done & confirmed that `gh` is still successfully authenticated.

File: .github/workflows/update-tor.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          GH_TOKEN: ${{ secrets.ISSUES_TOKEN }}
+        steps:
+            - name: Install dependencies
+              run: |
+                apt-get update && apt-get install --yes reprepro ca-certificates dctrl-tools \
+                  git git-lfs openssh-client python3 gh
+            - uses: actions/checkout@v4
+              with:
+                lfs: true
+                fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (eloquence) [reply .github/workflows/update-tor.yml]: I think you're right, I may have assumed this was related to proper LFS functioning based on the other workflow, but if it's needed for full branch/tag access, I don't think any of the code in this job requires it.

File: .github/workflows/update-tor.yml (hunk 3 above)
Comment (eloquence) [reply .github/workflows/update-tor.yml]: It was a timer, I just briefly switched it back just now to test my changes :) I'll set it to 7 AM

File: .github/workflows/update-tor.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (legoktm) [reply .github/workflows/update-tor.yml]: Ah, no, the other job needs it because it looks through git history to see what new files have been added.

**Review by legoktm — APPROVED** [review]
Awesome, thanks :D

**Pull request merged**